lib/tasks/gdpr_delete.rake

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+namespace :gdpr do
+  desc 'Deletes user accounts in accordance with GDPR. ' \
+       'Requires a text file with external user IDs (one per line) and ' \
+       'the corresponding consumer for deletion. Pass those via ' \
+       'TXT=/path/to/file.txt and CONSUMER=Name.'
+  task delete_users: :environment do
+    text_file_location = ENV['TXT']
+    consumer_name = ENV['CONSUMER']
+
+    if text_file_location.blank? || consumer_name.blank?
+      warn '--------------------------------------'
+      warn 'GDPR deletion tool to remove personal user data. USAGE:'
+      warn '--------------------------------------'
+      warn 'rake gdpr:delete_users TXT=./accounts.txt CONSUMER=openHPI'
+      exit 1
+    end
+
+    consumer = Consumer.where('lower(name) = ?', consumer_name.downcase).first
+
+    if consumer.blank?
+      warn '--------------------------------------'
+      warn "ERROR: The consumer '#{consumer_name}' was not found but is mandatory."
+      warn '--------------------------------------'
+      exit 2
+    end
+
+    file = begin
+      File.open(text_file_location, 'r')
+    rescue Errno => e
+      warn '--------------------------------------'
+      warn "ERROR: The file '#{text_file_location}' could not be opened: #{e.message}"
+      warn '--------------------------------------'
+      exit 3
+    end
+
+    unknown_user_ids = []
+    users_deleted = 0
+    file.each_line do |row|
+      next if row.blank?
+
+      user = ExternalUser.find_by(external_id: row.strip, consumer: consumer)
+      if user.blank?
+        unknown_user_ids << row
+        next
+      end
+
+      user.update!(name: 'Deleted User', email: nil)
```

Comment:
This will make the task fail if you cannot update the user. Is this intended? From the other error handling, I assume we'd like to continue and only add the user ID to the deleted users if we could update.

Reply:
Indeed, that's true: The task would fail. While I wouldn't expect that a user update cannot be performed, I agree that another error handling would be more suitable. Therefore, I changed it accordingly.